bundle/edu.gemini.lchquery.servlet/src/main/java/edu/gemini/lchquery/servlet/LchQueryFunctor.java

```diff
@@ -1,549 +0,0 @@
-
-    static {
-        dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
-    }
-
-    // Type of query results: only programs, progs + observations or progs + obs + targets
-    enum QueryType {
-        PROGRAMS, OBSERVATIONS, TARGETS
-    }
-
-    private QueryResult _queryResult;
-    private QueryType _queryType;
-
-    private String _programSemester;
-    private String _programTitle;
-    private String _programReference;
-    private String _programActive;
-    private String _programCompleted;
-    private String _programNotifyPi;
-    private String _programRollover;
-
-    private String _observationTooStatus;
-    private String _observationName;
-    private String _observationStatus;
-    private String _observationInstrument;
-    private String _observationAo;
-    private String _observationClass;
-
-    public LchQueryFunctor(QueryType queryType, String programSemester, String programTitle, String programReference,
-                           String programActive, String programCompleted, String programNotifyPi, String programRollover,
-                           String observationTooStatus, String observationName, String observationStatus, String observationInstrument,
-                           String observationAo, String observationClass) {
-        _queryResult = new QueryResult();
-        _queryResult.setProgramsNode(new ProgramsNode());
-
-        _queryType = queryType;
-
-        _programSemester = programSemester;
-        _programTitle = programTitle;
-        _programReference = programReference;
-        _programActive = programActive;
-        _programCompleted = programCompleted;
-        _programNotifyPi = programNotifyPi;
-        _programRollover = programRollover;
-
-        _observationTooStatus = observationTooStatus;
-        _observationName = observationName;
-        _observationStatus = observationStatus;
-        _observationInstrument = observationInstrument;
-        _observationAo = observationAo;
-        _observationClass = observationClass;
-    }
-
-    /**
-     * Called once per program by the <code>{@link edu.gemini.pot.spdb.IDBQueryRunner}</code>
-     * implementation.
-     */
-    public void execute(IDBDatabaseService database, ISPNode node, Set<Principal> principals) {
-        ISPProgram prog = (ISPProgram) node;
-        try {
-            if (_match(prog)) {
-                List<ISPObservation> obsList = new ArrayList<ISPObservation>();
-                for (ISPObservation obs : prog.getAllObservations()) {
-                    if (_match(prog, obs)) {
-                        obsList.add(obs);
-                    }
-                }
-                if (obsList.size() != 0) {
-                    if (_queryType == QueryType.PROGRAMS) {
-                        addProgram(prog, new ArrayList<ISPObservation>());
-                    } else {
-                        addProgram(prog, obsList);
-                    }
-                }
```

Comment:
Here's the corresponding [JIRA issue](http://jira.gemini.edu:8080/browse/REL-2929).